### PR TITLE
Define triangle_count C API

### PR DESCRIPTION
**Pull request by ChuckHastings**
This defines the C API for triangle counting.

This PR is independent of #2253 and can be merged independently.  The change here defines the C API and returns the `CUGRAPH_NOT_IMPLEMENTED` error when called.  Once #2253 is completed and merged, a follow-up PR will fill in the C API implementation for triangle counting (although the code is written and untested in this PR).

### CODE REVIEW

File: cpp/include/cugraph_c/community_algorithms.h
Comment (seunghwak) [inline cpp/include/cugraph_c/community_algorithms.h:40]: Add a comment saying "if this is NULL, the entire set of vertices is assumed"?

**Review by seunghwak — APPROVED** [review]
LGTM (except for one comment about documentation).

File: cpp/include/cugraph_c/community_algorithms.h
Comment (ChuckHastings) [reply cpp/include/cugraph_c/community_algorithms.h:40]: Will fix in next push.

File: cpp/include/cugraph_c/community_algorithms.h
Comment (rlratzel) [inline cpp/include/cugraph_c/community_algorithms.h:44]: Why is `result` labelled `in` but `error` is an `out`?  I figured they'd both be out.

**Review by rlratzel — APPROVED** [review]
LGTM. I focused on the public API part and less on the (commented out) implementation.  I had one questions which need not hold up approval.

File: cpp/include/cugraph_c/community_algorithms.h
Comment (ChuckHastings) [reply cpp/include/cugraph_c/community_algorithms.h:44]: Good catch.  Should both be out.  I will correct in a follow-up PR.

**Comment by ChuckHastings** [comment]
@gpucibot merge

**Pull request merged**
